pkg/streamingpromql/testutils/utils.go

```diff
@@ -20,6 +20,17 @@ import (
 // Why do we do this rather than require.Equal(t, expected, actual)?
 // It's possible that floating point values are slightly different due to imprecision, but require.Equal doesn't allow us to set an allowable difference.
 func RequireEqualResults(t testing.TB, expr string, expected, actual *promql.Result, skipAnnotationComparison bool) {
+
+	if expected == nil && actual == nil {
+		return
+	} else if expected == nil {
+		require.Fail(t, `expected nil but got type: %v`, actual)
+		return
+	} else if actual == nil {
+		require.Fail(t, "expected a value but got nil")
+		return
+	}
+
```

Comment:
Might be able to simplify this to:

```suggestion
	if expected == nil {
		require.Nil(t, actual, "expected nil result")
		return
	}
	
	require.NotNil(t, actual, "expected a result but got nil")
```